src/Dofs/DofHandler.jl

```diff
@@ -168,8 +168,15 @@ Return the number of degrees of freedom for the cell with index `cell`.
 
 See also [`ndofs`](@ref).
 """
-function ndofs_per_cell(dh::DofHandler, cell::Int=1)
-    @boundscheck 1 <= cell <= getncells(get_grid(dh))
+function ndofs_per_cell(dh::DofHandler)
+    if length(dh.subdofhandlers) > 1
+        error("There are more than one subdofhandler. Use `ndofs_per_cell(dh, cellid::Int)` instead.")
+    end
+    @assert length(dh.subdofhandlers) != 0
+    return @inbounds ndofs_per_cell(dh.subdofhandlers[1])
+end
+function ndofs_per_cell(dh::DofHandler, cell::Int)
```

Comment:
AFAIK the current implementation would never elude the boundscheck, so I would suggest
```suggestion
@propagate_inbounds function ndofs_per_cell(dh::DofHandler, cell::Int)
```
or 
```suggestion
@inline function ndofs_per_cell(dh::DofHandler, cell::Int)
```
or 
remove `@boundscheck` below. 

While I suppose that it is ok to rely on `dh.cell_to_subdofhandler[cell]` returning a valid index for `dh.subdofhandlers`, it is checked above for the other dispatch. 
In that respect, perhaps it would be better to just skip the boundscheck line, and remove the `@inbounds`?
Have you tried benchmarking if it makes a significant impact?

Reply:
One problem is that `dh.cell_to_subdofhandler[cell]` might return -1 when not all cells belong to a subdofhandler. This is extremely uncommon, but it was what caused the issue in L2Projector in the first case.

I dont really understand the reasoning in you comment (I dont know all the details of `@boundscheck` `@inbounds` etc)...What do you think about the latest commit?

Or is better to ignore bounds checking all together?
```
@inline function ndofs_per_cell(dh::DofHandler, cell::Int)
    return ndofs_per_cell(dh.subdofhandlers[dh.cell_to_subdofhandler[cell]])
end
```

Reply:
> ```julia
> @inline function ndofs_per_cell(dh::DofHandler, cell::Int)
>     return ndofs_per_cell(dh.subdofhandlers[dh.cell_to_subdofhandler[cell]])
> end
> ```

Yes, unless there is some major performance loss, I prefer this version (and can be optimized later with inbounds if required). With the latest commit, we anyways don't use inbounds on `dh.cell_to_subdofhandler[cell]`. 

(My reasoning is based on that inlining is required for `@boundscheck` to work as desired, at least if I understand the [docs](https://docs.julialang.org/en/v1/base/base/#Base.@boundscheck) correctly.)